### PR TITLE
add scuttle exceptions for JSON and Date

### PR DESCRIPTION
**Pull request by pedronfigueiredo**
## Explanation

As suggested in [this comment](https://github.com/MetaMask/metamask-extension/pull/18347/files#r1182496069), this PR makes changes to the build system atomically for increased visibility for the changelog.

## Screenshots of before the changes

Screenshot 2023-05-03 at 17 09 25
Screenshot 2023-05-03 at 17 02 26

### CODE REVIEW

**Review by brad-decker — DISMISSED** [review]

**Review by DDDDDanica — DISMISSED** [review]

File: development/build/index.js [hunk 1]
```diff
@@ -132,6 +132,8 @@ async function defineAndRunBuildTasks() {
         'stateHooks',
         'sentryHooks',
         'sentry',
+        'JSON',
```
Comment (Gudahtt) [inline development/build/index.js:135]: Do we know where these are being used? Are we sure it's related to Sentry?

Comment (Gudahtt) [reply development/build/index.js:135]: I'm guessing that they're used by some dependency not related to Sentry, in which case they'd be more appropriate to list under the "globals chrome driver needs to function (test env)" section

Comment (pedronfigueiredo) [reply development/build/index.js]: Not sure where these are being used, so I changed them to the other section as per your suggestion

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by DDDDDanica — APPROVED** [review]

**Pull request merged**